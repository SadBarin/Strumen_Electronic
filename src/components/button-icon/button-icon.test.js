import ButtonIcon from "./button-icon";

const elementProp = {
  icon: "bi-eyedropper",
  onClick: () => console.log('Simple test'),
  propsClass: 'test'
};

describe("Snapshot check props", () => {
  const element = (props) => render(<ButtonIcon {...props} />);

  it("LogicElement use Transferred Props", () => {
    const buttonIcon = element(elementProp)
    expect(toJson(buttonIcon)).toMatchSnapshot();
  });

  it("LogicElement use Default Props", () => {
    const buttonIcon = element()
    expect(toJson(buttonIcon)).toMatchSnapshot();
  });
});

describe("Check defaultProps", () => {
  it("Check default function onClick()", () => {
    const result = ButtonIcon.defaultProps.onClick()
    expect(result).toBeUndefined();
  });
});
