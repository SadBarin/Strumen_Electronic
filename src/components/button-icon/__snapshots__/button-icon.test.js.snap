// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot check props LogicElement use Default Props 1`] = `
<button
  class="btn text-white icon-button "
  type="button"
>
  <i
    class="bi bi-cpu"
  />
</button>
`;

exports[`Snapshot check props LogicElement use Transferred Props 1`] = `
<button
  class="btn text-white icon-button test"
  type="button"
>
  <i
    class="bi bi-eyedropper"
  />
</button>
`;
